src/users_router.rs

```diff
@@ -79,10 +79,10 @@ struct LoginResponse {
 }
 
 impl LoginResponse {
-    fn new(user: &User) -> IronResult<Response> {
-        let session_token = match SessionToken::new(&user) {
+    fn with_user(user: &User) -> IronResult<Response> {
```

Comment:
Another `new` issue. This lead me to think we should be implementing this as traits. I need to read about how to effectively organize and architect in Rust.